.circleci/config.yml

```diff
@@ -662,14 +666,27 @@ jobs:
           name: Build go-ffi
           command: just build-go-ffi
           working_directory: packages/contracts-bedrock
+      - when:
+          condition:
+            equal: ["<<parameters.test_upgrade>>", "true"]
+          steps:
+            - restore_cache:
+                name: Restore forked state
+                key: forked-state-contracts-bedrock-tests-upgrade
       - run:
           name: Run tests
           command: |
-            TEST_FILES=$(<<parameters.test_list>>)
-            TEST_FILES=$(echo "$TEST_FILES" | circleci tests split --split-by=timings)
-            TEST_FILES=$(echo "$TEST_FILES" | sed 's|^test/||')
-            MATCH_PATH="./test/{$(echo "$TEST_FILES" | paste -sd "," -)}"
-            forge <<parameters.test_command>> <<parameters.test_flags>> --match-path "$MATCH_PATH"
+            if [[ "$(<<parameters.test_upgrade>>)" == "true" ]]; then
+              echo "Running upgrade tests"
+              export ETH_RPC_URL="https://ci-mainnet-l1-archive.optimism.io"
+              just test-upgrade
```

Comment:
unlike the non-upgrade path, here we use the `just` command, which does its own filtering to limit tests to L1 contracts.

Reply:
We don't use test splitting anymore anyway because it all runs on one box. I'd recommend just making this part of a new job.